packages/ui/src/Toggle.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC, FocusEvent, ChangeEvent, AriaAttributes, useRef } from 'react'
+import cn from 'classnames'
+import { errorId } from './Error'
+// import { Error, errorId } from './Error'
+import { Help, helpTooltipId } from './Help'
+import { useUID } from 'react-uid'
+import { DataTestProp } from '@hazelcast/helpers'
+import styles from './Toggle.module.scss'
+
+export type ToggleCoreProps = {
+  name: string
+  value?: string
+  checked?: boolean
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void
+  error?: string
+
+  label: string | React.ReactNode
+  helperText?: string
+  disabled?: boolean
+  className?: string
+  classNameLabel?: string
+}
+
+export type ToggleProps = ToggleCoreProps & DataTestProp
+
+export const Toggle: FC<ToggleProps> = ({
+  name,
+  value,
+  onChange,
+  onBlur,
+  error,
+  checked,
+  label,
+  helperText,
+  disabled,
+  className,
+  classNameLabel,
+  'data-test': dataTest,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null)
+  const id = useUID()
+
+  let ariaChecked: AriaAttributes['aria-checked'] = checked ? 'true' : 'false'
+
+  return (
+    <div className={cn(styles.wrapper, className)} data-test={dataTest}>
+      <input
+        type="checkbox"
+        ref={inputRef}
+        id={id}
+        name={name}
+        checked={!!checked}
+        onChange={onChange}
+        onBlur={onBlur}
+        value={value}
+        disabled={disabled}
+        aria-checked={ariaChecked}
+        aria-invalid={!!error}
+        aria-describedby={helperText && helpTooltipId(id)}
+        aria-errormessage={error && errorId(id)}
+      />
+
+      <label className={cn(classNameLabel, {[styles.disabled]: disabled})} htmlFor={id}>
```

Comment:
Couldn't we use our Label?

Reply:
unfortunately :/ i'm abusing the internal label element to display a track there.

Reply:
`Label` accepts `ReactElement` as `label`. Couldn;t you pass anything you want, including your toggle track?